src/scripts/install_machine.py

```diff
@@ -0,0 +1,326 @@
+#! /usr/bin/python3
+
+import traceback
+import json
+import subprocess
+import sys
+import os
+import xml.etree.ElementTree as ET
+import tempfile
+import logging
+
+
+def virsh(connection, *args):
+    cmd = ("virsh", "-c", f"qemu:///{connection}") + args
+    logging.debug("Running virsh command :" + ' '.join(cmd))
+    return subprocess.check_output(cmd)
+
+
+def assert_vm_exists(connection, name):
+    logging.debug(virsh(connection, 'list', '--all', '--name').strip().split())
+    assert name.encode() in virsh(connection, 'list', '--all', '--name').strip().split()
+
+
+def get_graphics_capabilies(connection):
+    capabilities = virsh(connection, 'domcapabilities')
+
+    root = ET.ElementTree(ET.fromstring(capabilities))
+    graphics = root.find('devices').find('graphics')
+    supported = graphics.get('supported')
+
+    consoles = []
+    if supported == 'yes':
+        for value in graphics.find('enum').findall('value'):
+            consoles.append(value.text)
+
+    logging.debug('get_graphics_capabilies: ' + ', '.join(consoles))
+
+    return list(filter(lambda console: console in ['vnc', 'spice'], consoles))
+
+
+def prepare_unattended(params, args, files):
+    if args.get('unattended'):
+        params.append("--unattended")
+
+        unattended_params = []
+        unattended_params.append(f"profile={args['profile']}")
+
+        if 'rootPassword' in args:
+            with tempfile.NamedTemporaryFile(
```

Comment:
Right now this can't work, as the temporary file will get cleaned up before calling virt-install (i.e. as soon as the `with` block ends). (but you are already discussing this in the thread with Lis)

Reply:
It works because of the `delete=False`, but it's very much missing the point of using it as a context manager.

Reply:
Fixed with the contextmanager approach. It worked because of the delete=False.